pkg/gitops/flux/installer.go

```diff
@@ -143,17 +143,21 @@ func (fi *Installer) Run(ctx context.Context) (string, error) {
 	instruction := fmt.Sprintf("please configure %s so that the following Flux SSH public key has write access to it\n%s",
 		fi.opts.Repo.URL, fluxSSHKey.Key)
 
-	gitProvider, err := providers.GetProvider(fi.cfg.Git.Repo.URL)
+	client, err := deploykey.GetDeployKeyClient(ctx, fi.cfg.Git.Repo.URL)
 	if err != nil {
-		logger.Warning("could not find git provider implementation for url %q: %q. Skipping authorization of SSH key", fi.cfg.Git.Repo.URL, err.Error())
+		logger.Warning(
+			"could not find git provider implementation for url %q: %q. Skipping authorization of SSH key",
+			fi.cfg.Git.Repo.URL,
+			err.Error(),
+		)
 		return instruction, nil
 	}
 
 	keyTitle := KeyTitle(*fi.cfg.Metadata)
-	err = gitProvider.AuthorizeSSHKey(ctx, key.SSHKey{
-		Title:    keyTitle,
-		Key:      fluxSSHKey.Key,
-		ReadOnly: fi.cfg.Git.Operator.ReadOnly,
+	_, err = client.Create(ctx, gitprovider.DeployKeyInfo{
```

Comment:
do you think you want to `Reconcile` here, or is it an error if a key with that same title already would exist?